### DISCUSSION TITLE
Configuration problem

### DISCUSSION THREAD

**Discussion by jaromirj** [E3SM-Project/E3SM#6087] [E3SM model help]
Dear Developers,
I am tryting to configure E3SM on new machine. It is Tryton Plus located in the CI TASK (Gdansk, Poland). I have modified the files config_machine.xml and config_batch.xml and I am not able to pass setup of my case (./case.setup command). The main command I run is:

./create_newcase -case E3SMv2.1Gtest --compset GMPAS-JRA --res TL319_oEC60to30v3 -mach trytonp

then I enter to the case directory and execute command ./case.setup and get an error:
```
Traceback (most recent call last):
  File "./case.setup", line 106, in <module>
    _main_func(__doc__)
  File "./case.setup", line 102, in _main_func
    case.case_setup(clean=clean, test_mode=test_mode, reset=reset, keep=keep)
  File "...E3SMv2/cime/CIME/case/case_setup.py", line 499, in case_setup
    is_batch=is_batch,
  File ".E3SMv2/cime/CIME/utils.py", line 2477, in run_and_log_case_status
    rv = func()
  File "...E3SMv2/cime/CIME/case/case_setup.py", line 462, in <lambda>
    self, caseroot, clean=clean, test_mode=test_mode, reset=reset, keep=keep
  File "...E3SMv2/cime/CIME/case/case_setup.py", line 290, in _case_setup_impl
    case.load_env()
  File "...E3SMv2/cime/CIME/case/case.py", line 2157, in load_env
    self._loaded_envs = env_module.load_env(self, job=job, verbose=verbose)
  File "...E3SMv2/cime/CIME/XML/env_mach_specific.py", line 140, in load_env
    modules_to_load, force_method=force_method, verbose=verbose
  File "...E3SMv2/cime/CIME/XML/env_mach_specific.py", line 169, in _load_modules
    self._load_module_modules(modules_to_load, verbose=verbose)
  File "...E3SMv2/cime/CIME/XML/env_mach_specific.py", line 463, in _load_module_modules
    exec(py_module_code)
  File "<string>", line 1
    'tryton/mkl/2019' load complete.
                                      ^
SyntaxError: invalid syntax

```
I think I have PE problem, however i am not able to solve it myself. Could you be so kind and try to give me direction how to continue work with this system please?
All the best
Jaromir

**Comment by jaromirj** [comment]
Hello again,
I should add that I have been working on ice-ocean models for a years in the Institute of Oceanology, Poland. I have experience with CICE, POP2, CESM, RASM (RACM) and ROMS.
thank you in advance for your help
Jaromir Jakacki

**Comment by jonbob** [comment]
Hi Dr. Jakacki,

My name is Jon Wolfe and I'm one of the E3SM software engineers.

It might help if you post the changes you've made to support this new machine -- your create_newcase command is otherwise fine.

Thanks,
Jon

**Comment by jaromirj** [comment]
I do not know why it skipped brackets in the first copy - will try to fix it:
```
  <machine MACH="trytonp">
    <DESC> TASK intel tryton plus cluster 24-cores per node</DESC>
    <OS>LINUX</OS>
    <COMPILERS>intel</COMPILERS>
    <MPILIBS>mpi</MPILIBS>
    <PROJECT>pt00625</PROJECT>
    <SAVE_TIMING_DIR>$CIME_OUTPUT_ROOT/timing</SAVE_TIMING_DIR>
    <SAVE_TIMING_DIR_PROJECTS>pt00625</SAVE_TIMING_DIR_PROJECTS>
    <CIME_OUTPUT_ROOT>/users/project1/pt00625/e3sm/$CASE</CIME_OUTPUT_ROOT>
    <DIN_LOC_ROOT>/users/scratch1/jakacki/pt00625/e3sm/inputdata</DIN_LOC_ROOT>
    <DIN_LOC_ROOT_CLMFORC>$DIN_LOC_ROOT/atm/datm7</DIN_LOC_ROOT_CLMFORC>
    <DOUT_S_ROOT>$CIME_OUTPUT_ROOT/archive/$CASE</DOUT_S_ROOT>
    <BASELINE_ROOT>$CIME_OUTPUT_ROOT/$COMPILER</BASELINE_ROOT>
    <CCSM_CPRNC>$DOUT_S_ROOT</CCSM_CPRNC>
    <GMAKE_J>8</GMAKE_J>
    <TESTS>e3sm_developer</TESTS>
    <BATCH_SYSTEM>slurm</BATCH_SYSTEM>
    <SUPPORTED_BY>jjakacki at iopan dot pl</SUPPORTED_BY>
    <MAX_TASKS_PER_NODE>24</MAX_TASKS_PER_NODE>
    <MAX_MPITASKS_PER_NODE>24</MAX_MPITASKS_PER_NODE>
    <mpirun mpilib="default">
      <executable>mpirun</executable>
      <arguments/>
    </mpirun>
    <module_system type="module" allow_error="true">
      <init_path lang="perl">/apl/tryton/modules/4.6.0/init/perl</init_path>
      <init_path lang="python">/apl/tryton/modules/4.6.0/init/python</init_path>
        <cmd_path lang="perl">module</cmd_path>
        <cmd_path lang="python">module</cmd_path>
        <cmd_path lang="sh">module -q</cmd_path>
        <cmd_path lang="csh">module -q</cmd_path>
      <modules>
        <command name="load">tryton/python-intel/2019</command>
        <command name="load">tryton/netcdf/4.7.4-intel19-mpi-iopan</command>
      </modules>
    </module_system>
    <RUNDIR>$CIME_OUTPUT_ROOT/run</RUNDIR>
    <EXEROOT>$CIME_OUTPUT_ROOT/bld</EXEROOT>
    <environment_variables>
      <env name="NETCDF_HOME">$ENV{NETCDF_ROOT}</env>
      <env name="MKL_PATH">$ENV{MLIBHOME}</env>
    </environment_variables>
  </machine>

```

**Reply by rljacob** [reply]
You have to wrap it in code block formatting.

**Comment by jaromirj** [comment]
I had to take a snapshot - I don't know why it loses configuration
![Screenshot 2023-11-29 at 21 33 33](https://github.com/E3SM-Project/E3SM/assets/7846448/f114c548-b7ef-49c7-8d0c-de431c1c9d08)

it is attached

**Reply by jonbob** [reply]
Are you familiar with git? Can you post the output from "git status" and from "git diff" for any file the status says has changed?

**Reply by rljacob** [reply]
What version of python do you get with that tryton/python-intel/2019 module?  You need at least 3.10.

**Comment by jaromirj** [comment]
Thanks again for you effort. I am including pictures and the file. Everything should be visible.
![Screenshot 2023-11-30 at 07 23 18](https://github.com/E3SM-Project/E3SM/assets/7846448/5b699006-55d3-4aa1-9faf-1dbd4e2052b9)
![Screenshot 2023-11-30 at 07 23 51](https://github.com/E3SM-Project/E3SM/assets/7846448/1a39f75f-8798-40cb-9298-e0296f09d56e)
[cmach.xml.diff.txt](https://github.com/E3SM-Project/E3SM/files/13509252/cmach.xml.diff.txt)

:)

**Comment by jaromirj** [comment]
The version of python is 3.6.9. I added two lines in env_mach_specific.py:
465             print("Python version")
466             print (sys.version)
 and it printed the it out. I am able to reproduce it of course.
Jaromir

**Reply by rljacob** [reply]
You'll need to get the sysadmins for that system to install a module with python 3.10 or later.  If they don't already have one.

**Reply by jaromirj** [reply]
I know there is 3.11 version. I will try to use it.
thanks a lot
Jaromir

**Comment by jaromirj** [comment]
Next time I will try to block the formatting when I try to paste. I have no experience with this system.
I am very sorry for that.
Jaromir
